cli/cmd/inject.go

```diff
@@ -156,6 +158,15 @@ func injectDaemonSet(bytes []byte) (interface{}, error) {
 	}, nil
 }
 
+func containsProxyAPIContainer(spec v1.PodSpec, containerName string) bool {
```

Comment:
I think this name isn't quite right.  This method seems to be a general way to test if a pod contains a named container, not just the Proxy API container.

Reply:
Thanks Alex, good suggestion. However, we no longer need this function with the new refactor in an upcoming commit. In this commit, we no longer need to search for the proxy container within a deployment spec. We can safely assume that if we read in the YAML object and the object is a deployment, we check for the deployment name to detect if it is the control plane. If it is, we can then pass in the DNS name override.